### PR TITLE
build 0.10.0

### PR DESCRIPTION
**Pull request by boldorider4**
* [x] update dependencies from [upstream](https://github.com/pypa/build/blob/0.10.0/pyproject.toml)
* [x] checked [license](https://github.com/pypa/build/blob/0.10.0/LICENSE) file

other:
* updated authors
* fixed about section
* fixed build script
* added pip check
* fixed self packaging test
* commented skipped tests

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,37 +1,61 @@
-{% set version = "0.7.0" %}
-{% set sha256 = "1aaadcd69338252ade4f7ec1265e1a19184bf916d84c9b7df095f423948cb89f" %}
+{% set version = "0.10.0" %}
 
 package:
   name: build
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/b/build/build-{{ version }}.tar.gz
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:9]: this https://pypi.io/packages/source/b/build/build-0.10.0.tar.gz still exists, why from github ?

**Review by akabanovs — APPROVED** [review]
Can't spot any problems. LGTM

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM, just one question about source url

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:9]: Yes, please use the PyPI sdist.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,37 +1,61 @@
...
     - pyproject-build = build.__main__:entrypoint
 
 requirements:
+  build:
+    - git  # [not win]
   host:
-    - python >=3.6
+    - python
     - pip
+    - wheel
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
Wheel is not needed. `build` uses `flit-core`, which creates wheels without using the `wheel` package. Ignore the linter warning. The linter will be fixed in https://github.com/anaconda-distribution/anaconda-linter/pull/272.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,37 +1,61 @@
...
-    - pep517 >=0.9.1
-    - tomli
-    - importlib-metadata
+    - python
+    - packaging >=19.0
+    - pyproject_hooks
+    - importlib-metadata >=0.22  # [py<38]
+    - tomli >=1.1.0  # [py<311]
+
+{% set tests_to_skip = "test_build_wheel or test_build_sdist" %}
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Why do we skip these tests?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,37 +1,61 @@
...
+{% set tests_to_skip = tests_to_skip + " or test_init" %}  # [linux]
+{% set tests_to_skip = tests_to_skip + " or test_output_env_subprocess_error" %}  # [not win]
 
 test:
+  source_files:
+    - src/build/
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Why is `src/build` needed? These files should already be installed in the package...

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (boldorider4) [reply recipe/meta.yaml]: OK!

File: recipe/meta.yaml (hunk 3 above)
Comment (boldorider4) [reply recipe/meta.yaml]: Forgot to add a couple of comment. Will add them.

File: recipe/meta.yaml (hunk 4 above)
Comment (boldorider4) [reply recipe/meta.yaml]: I remember trying to run the tests without these files and they wouldn't run.
Furthermore, [this](https://github.com/pypa/build/blob/cd06da25481b9a610f846fa60cb67b5a5fa9a051/pyproject.toml#L80C1-L80C1).
I will try again without the build dir and see if they indeed don't run.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: https://github.com/pypa/build/blob/cd06da25481b9a610f846fa60cb67b5a5fa9a051/pyproject.toml#L80 is only for code coverage purposes. We don't run tests with coverage enabled.

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml:9]: The problem with the PyPi is that it does not include the `tests` dir. I wrote an [issue](https://github.com/pypa/build/issues/656) about it. If you (@M-Waszkiewicz-Anaconda, @JeanChristopheMorinPerso ) know a better suited URL, let me know.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:9]: In this case, you can download the GH tarball, for example https://github.com/pypa/build/archive/refs/tags/0.10.0.tar.gz.

Comment (boldorider4) [reply recipe/meta.yaml:9]: Thanks!

File: recipe/meta.yaml (hunk 4 above)
Comment (boldorider4) [reply recipe/meta.yaml]: let me try it one time without it and then I'll remove it.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**